### DISCUSSION TITLE
pr template

### DISCUSSION THREAD

**Discussion by aviv82** [lab-antwerp-1/group4-soc-a-list#19] [General]
# pr description

## Checklists

- [ ] All CI checks pass
- [ ] The branch works when you pull it and run it locally

### HTML

- [ ] the code is well-formatted
- [ ] the HTML code passes [validator.w3.org](https://validator.w3.org/)
- [ ] there are no inline styles (example: `style='color: red;'`)
- [ ] there are no `<style>` tags with CSS, all styles are hrefs
- [ ] there is no inline JavaScript (example: `onclick='doSomething()'`)
- [ ] there are no `<script>` tags with JS, all JS is in an separate file
- [ ] `id`s are used for JavaScript only, not for CSS
- [ ] semantic tags are used
- [ ] ARIA labels are used where necessary
- [ ] spelling and grammar is correct in all site content

### CSS

- [ ] the code is well-formatted
- [ ] the code does not use any `#` id's
- [ ] styles are responsive

### JavaScript

- [ ] code is clean and easy to read
- [ ] helpful variable names are used
- [ ] all comments are clear and help to understand the code
- [ ] there is no unused code in comments
- [ ] all file names are helpful and match their exports

#### src/data.js

- [ ] the file does not import anything
- [ ] the file only exports JS data (primitives, arrays, objects, ...)
- [ ] the file does not contain any functions or logic
- (optional) the data is documented with JSDocs

#### src/init

- [ ] there is an `index.js` that is included by the `index.html` file
- [ ] `../listener` files are imported
- [ ] there is no code that needs to run _after_ the page is initialized
- [ ] any other files have helpful names

#### src/listeners

- [ ] the DOM can be queried to find elements
- [ ] the DOM _is not_ be modified
- [ ] handlers are imported and used as callbacks to event listeners
- [ ] no other functions are imported or used
- [ ] the file name makes sense for the listener

#### src/handlers

- [ ] handlers have a JSDoc comment
- [ ] the function name matches the file name
- [ ] handlers are used as a callback to `.addEventListener` somewhere in the program
- [ ] functions from `/logic` or `/procedures` files may be imported and called
- [ ] functions from `/handlers` may be imported and attached to the DOM with event listeners
- [ ] data from `/data.js` may be imported and used
- [ ] handlers may read and write to the DOM
- [ ] handlers _do not_ return values that you will need later in the program

#### src/components

- [ ] A DOM element is returned
- [ ] handlers can be imported and used as callbacks to event listeners
- [ ] _logic_, _components_ and _procedures_ can also be imported
- [ ] there is an HTML test file to render the component with different inputs
- [ ] each component has a `.test.html` file
- [ ] bonus points for a unit test `.spec.js` file

#### src/logic

- [ ] the handler has a complete and correct JSDoc comment
- [ ] the function name matches the file name
- [ ] other functions from `/logic` may be imported and called
- [ ] the function _does not_ read or write to the DOM
- [ ] the function _does not_ log to the console
- [ ] each logic function has a `.spec.js` file
  - [ ] tests are simple with no extra code
  - [ ] tests are well-named
  - [ ] tests cover common use cases (bonus for edge cases)
  - [ ] tests check for side-effects (if necessary)

**Comment by AlinaTaoRao** [comment]
# feature remove button: version 2
close #16 


## Checklists

- [x] All CI checks pass
- [x] The branch works when you pull it and run it locally



### HTML

- [x] the code is well-formatted
- [x] the HTML code passes [validator.w3.org](https://validator.w3.org/)
- [x] there are no inline styles (example: `style='color: red;'`)
- [x] there are no `<style>` tags with CSS, all styles are hrefs
- [x] there is no inline JavaScript (example: `onclick='doSomething()'`)
- [x] there are no `<script>` tags with JS, all JS is in an separate file
- [x] `id`s are used for JavaScript only, not for CSS
- [x] semantic tags are used
- [x] ARIA labels are used where necessary
- [x] spelling and grammar is correct in all site content

### CSS

- [x] the code is well-formatted
- [x] the code does not use any `#` id's
- [x] styles are responsive

### JavaScript

- [x] code is clean and easy to read
- [x] helpful variable names are used
- [x] all comments are clear and help to understand the code
- [x] there is no unused code in comments
- [x] all file names are helpful and match their exports

#### src/data.js

- [x] the file does not import anything
- [x] the file only exports JS data (primitives, arrays, objects, ...)
- [x] the file does not contain any functions or logic
- (optional) the data is documented with JSDocs

#### src/init

- [x] there is an `index.js` that is included by the `index.html` file
- [x] `../listener` files are imported
- [x] there is no code that needs to run _after_ the page is initialized
- [x] any other files have helpful names

#### src/listeners

- [x] the DOM can be queried to find elements
- [x] the DOM _is not_ be modified
- [x] handlers are imported and used as callbacks to event listeners
- [x] no other functions are imported or used
- [x] the file name makes sense for the listener

#### src/handlers

- [x] handlers have a JSDoc comment
- [x] the function name matches the file name
- [x] handlers are used as a callback to `.addEventListener` somewhere in the program
- [x] functions from `/logic` or `/procedures` files may be imported and called
- [x] functions from `/handlers` may be imported and attached to the DOM with event listeners
- [x] data from `/data.js` may be imported and used
- [x] handlers may read and write to the DOM
- [x] handlers _do not_ return values that you will need later in the program

#### src/components

- [x] A DOM element is returned
- [x] handlers can be imported and used as callbacks to event listeners
- [x] _logic_, _components_ and _procedures_ can also be imported
- [x] there is an HTML test file to render the component with different inputs
- [x] each component has a `.test.html` file
- [x] bonus points for a unit test `.spec.js` file

#### src/logic

- [x] the handler has a complete and correct JSDoc comment
- [x] the function name matches the file name
- [x] other functions from `/logic` may be imported and called
- [x] the function _does not_ read or write to the DOM
- [x] the function _does not_ log to the console
- [x] each logic function has a `.spec.js` file
  - [x] tests are simple with no extra code
  - [x] tests are well-named
  - [x] tests cover common use cases (bonus for edge cases)
  - [x] tests check for side-effects (if necessary)
